test/sql/cast/test_string_cast.test

```diff
@@ -54,6 +54,15 @@ SELECT 2::DOUBLE::VARCHAR, 0.5::DOUBLE::VARCHAR, (-128.5)::DOUBLE::VARCHAR
 0.5
 -128.5
 
+query IIIII
+SELECT '0xF'::INTEGER, '0x0'::INTEGER, '0xFEE'::INTEGER, '0xfee'::INTEGER, '0x00FEE'::INTEGER
```

Comment:
Could we add some tests with:

* `'0x'::INT` (empty hex: should fail)
* `'0'::INT` (should succeed and return 0)
* Upper-case `x` (`'0XFF'::INT`)
* Mix of different cases (`'0XfEfE'::INT`)
* `'0xFFFFFFFFFFFFFFFFF'::INT` (i.e. very large overflow)
* `'0x000000000000000'::INT` (many zeros - should succeed)
* Exact bound tests for all types - should all succeed (`'0x7F'::TINYINT`, `'0x7FFF'::SMALLINT`, `'0x7FFFFFFF'::INT`, `'0x7FFFFFFFFFFFFFFF'::BIGINT`)
* Exact bound tests exceeded for all types - should all fail (`'0x80'::TINYINT`, `'0x8000'::SMALLINT`, `'0x80000000'::INT`, `'0x8000000000000000'::BIGINT`)
* `TRY_CAST` with an out-of-bounds integer (e.g. `SELECT TRY_CAST('0x80' AS TINYINT)` - should return `NULL`)
* The same exact bound tests with casts to unsigned types (`'0xFF'::UTINYINT`, etc)

Reply:
Yep, happy to add a chunk more tests given the general approach looks good :) I think I can see the right way of testing out of bounds (other than the try_cast which I'll do as well) - other files have "statement error", is that the right approach? Or should I be adding those cases to the c++ test cases?

Reply:
`statement error` is indeed the correct approach

Note that you should only test one out-of-bounds per query, otherwise the query will fail on the first error on not correctly test the remaining, so e.g.:

```sql
statement error
SELECT '0x80'::TINYINT;

statement error
SELECT '0x8000'::SMALLINT;

...
```